common/src/crypto.rs

```diff
@@ -32,20 +44,45 @@ impl fmt::Display for Hash256 {
 /// A cryptographic signature.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
 pub struct Signature {
-    pub dummy: String,
+    signature: Vec<u8>,
 }
 
 impl Signature {
     /// Creates a new signature from the given data and keys.
-    pub fn sign(_data: Hash256, _public_key: &PublicKey, _private_key: &PrivateKey) -> Self {
-        Signature {
-            dummy: "".to_string(),
-        }
+    pub fn sign(
+        data: Hash256,
```

Comment:
Why do we sign only hashes but [u8] messages?

Reply:
I think so.